pynamodb/indexes.py

```diff
@@ -43,14 +44,13 @@ def __init__(self) -> None:
             raise ValueError("No projection defined, define a projection for this class")
 
     def __set_name__(self, owner: Type[_M], name: str):
-        if not hasattr(self.Meta, "model"):
-            self.Meta.model = owner
+        if not hasattr(type(self), "_model"):
```

Comment:
Do we still need this? Why wouldn't `_initialize_indexes()` set it?

Reply:
good catch -- i don't think we do now since we never need to access the variable on a class